lib/api/apiUtils/object/storeObject.js

```diff
@@ -26,7 +26,9 @@ function checkHashMatchMD5(stream, hashedStream, dataRetrievalInfo, log, cb) {
             contentMD5,
         });
         log.trace('contentMD5 does not match, deleting data');
```

Comment:
Maybe I misunderstood what is the md5 here, in light of Lauren's comment, if the MD5 is the one coming from the user, we should not log error then since we do not want a malicious user to flood the server with error logs. But it can still be a good idea to merge the log lines into a single one even if they are debug logs.